llvm/lib/Target/AIE/AIELegalizerInfo.cpp

```diff
@@ -757,6 +759,23 @@ bool AIELegalizerInfo::legalizeG_UNMERGE_VALUES(LegalizerHelper &Helper,
   return true;
 }
 
+bool AIELegalizerInfo::legalizeG_SEXT_INREG(LegalizerHelper &Helper,
+                                            MachineInstr &MI) const {
+
+  MachineIRBuilder &MIRBuilder = Helper.MIRBuilder;
+  MachineRegisterInfo &MRI = *MIRBuilder.getMRI();
+
+  const Register DestReg = MI.getOperand(0).getReg();
+  const LLT DestRegTy = MRI.getType(DestReg);
+  const LLT S32 = LLT::scalar(32);
+
+  const int64_t Imm = MI.getOperand(2).getImm();
+  if ((Imm != 8 && Imm != 16) || DestRegTy != S32)
+    Helper.lowerSextInreg(MI);
+
+  return true;
+}
```

Comment:
CHECK: if DestRegTy > S32, the lowering will depend on existing legalization of the 64 bit shifts that are produced.
DestRegTy < S32  does not occur since it has been widened earlier?

Reply:
For the first case, this is covered by the test https://github.com/Xilinx/llvm-aie/blob/aeb9129715bd6baebf12b6b5da5893cb38811fe5/llvm/test/CodeGen/AIE/GlobalISel/legalize-sext-inreg.mir#L102C26-L102C47 . For the second case I believe yes, we did not clamp before, now the behavior is the same.